bundles/org.openhab.binding.linky/src/main/java/org/openhab/binding/linky/internal/handler/LinkyHandler.java

```diff
@@ -105,13 +105,12 @@ public LinkyHandler(Thing thing, LocaleProvider localeProvider, Gson gson, HttpC
         });
 
         this.cachedPowerData = new ExpiringDayCache<>("power cache", REFRESH_FIRST_HOUR_OF_DAY, () -> {
-            LocalDate to = LocalDate.now();
-            LocalDate from = to.minusDays(1);
-            Consumption consumption = getPowerData(from, to);
+            LocalDate today = LocalDate.now();
+            Consumption consumption = getPowerData(today.minusDays(2), today);
```

Comment:
A few questions:
- Do you need the data from the day before yesterday?
- Do you know when new data is available?

If the data is not needed, it seems that too much is requested as a work-around. Two considerations:
- If the data you actually need is not yet available, perhaps the HTTP code should be handled gracefully? I assume old data will not fix anything but the unhandled exception/HTTP code?
- Will new data arrive right after midnight in local time (currently UTC+2 in France) or at UTC date change? Currently `LocalDate` is used, so date will change at 22:00:00 UTC.

Reply:
No, we don't need the data from the day before yesterday, I requested it just to avoid the INTERNAL_SERVER_ERROR. Note that it is absolutely not a massive amount of data.
No, I don't know when each data will be available, this is at a certain time in the morning. That's why, the binding tries every 2 hours and cache the data as soon as it is properly returned. Some data are even available before other data but it is very variable and not predictable. Generally, at 7 in the morning, the binding got all its data but it can be earlier in the morning sometimes. Our idea was to get them as soon as possible.

Normally, when the value is not yet available for yesterday, the API returns data with a NaN value for that day.
But if we request the power peak only for yesterday, there is apparently a bug in the API and the API returns an INTERNAL_SERVER_ERROR rather than the data with a NaN value. By requesting two days, the API is returning NaN for yesterday, as expected.

So, yes, this is clearly a workaround to avoid an API bug but it looks to me as a clean and simple solution to avoid this API bug. I am going to add a comment in the code to explain that.

PS: that would be cool if that could be merged today, before the new milestone is generated.

Reply:
Comment added in the code.